ACMILabs/playlist-label
# How we tell our stories design

Resolves #61

I've added an alternative template for How We Tell Our Stories -type labels, and refactored the javascript to support this, and to allow key controls for debugging.

### Relevant design files
<img width="1067" alt="Layout" src="https://user-images.githubusercontent.com/55459/89993269-c8958200-dcc9-11ea-8be3-973c2f56f2cd.png">

### Testing with local XOS
1. Set up a playlist for playlist label containing max 7 items, with videos and labels. You may want to make some works indigenous and have a title_annotation. You way want to put markdown in some of the description fields. Note the ID and an auth token.
2. edit `config.env` to include:
```
XOS_API_ENDPOINT=http://[YOUR_IP]:8000/api/
AUTH_TOKEN=[YOUR AUTH TOKEN]
XOS_MEDIA_PLAYER_ID=1
XOS_PLAYLIST_ID=[PLAYLIST_ID]
LABEL_TEMPLATE=up_next.html
```
3. Run `cd development && docker-compose up` to run the playlist app
4. View the label at localhost:8081 - ensure your labels are shown and design looks like the attached. Ensure the durations of your videos are properly reflected.
5. Press 0-9 to move the progress bar to 0-90% and see the times update. Press 't' to simulate a tap.
6. Press right arrow to move to next item. Notice that the browser # anchor updates with label ID. Refresh the browser to see that the current label is reloaded (albeit with flash of initial label - removing that can be for another ticket).
7. Remove `LABEL_TEMPLATE` env var, and restart the docker-compose command, and see that the standard label renders.

### Testing with playlist hardware (needs VPN connection to XOS)
1. Set up a playlist in XOS; note the ID.
2. Deploy a playlist label to use the `s__playlist-label-pi-4` app. Configure the device with `XOS_PLAYLIST_ID`, and `LABEL_TEMPLATE` as above.
3. Observe that the playlist works as expected.
4. If a keyboard is connected, the keys work as above.
5. If a media player is connected, the playlist updates to match
6. If a lens reader is connected, the lens reader captures the currently-shown label and the 'collect' animation displays properly.
